art/attacks/evasion/over_the_air_flickering/over_the_air_flickering_pytorch.py

```diff
@@ -0,0 +1,360 @@
+        :param round_samples: Granularity of the input values to be enforced if > 0.0.
+        """
+        super().__init__(estimator=classifier)
+
+        self.eps_step = eps_step
+        self.max_iter = max_iter
+        self.beta_0 = beta_0
+        self.beta_1 = beta_1
+        self.beta_2 = beta_2
+        self.loss_margin = loss_margin
+        self.batch_size = batch_size
+        self.start_frame_index = start_frame_index
+        self.num_frames = num_frames
+        self.round_samples = round_samples
+        self.end_frame_index = (
+            self.start_frame_index + self.num_frames if self.num_frames is not None else self.num_frames
+        )
+        self._check_params()
+
+    def generate(self, x: np.ndarray, y: Optional[np.ndarray] = None, **kwargs) -> np.ndarray:
+        """
+        Generate adversarial examples.
+
+        :param x: Original input samples.
+        :param y: Target values (class labels) one-hot-encoded of shape (nb_samples, nb_classes) or indices of shape
+                  (nb_samples,).
+        :return: Adversarial examples.
+        """
+        import torch  # lgtm [py/repeated-import]
+
+        y = check_and_transform_label_format(y, self.estimator.nb_classes)
+        if y is None:
+            logger.info("Using model predictions as true labels.")
+            y = get_labels_np_array(self.estimator.predict(x, batch_size=self.batch_size))
+
+        dataset = torch.utils.data.TensorDataset(
+            torch.from_numpy(x.astype(ART_NUMPY_DTYPE)),
+            torch.from_numpy(y.astype(ART_NUMPY_DTYPE)),
+        )
+
+        data_loader = torch.utils.data.DataLoader(
+            dataset=dataset, batch_size=self.batch_size, shuffle=False, drop_last=False
+        )
+
+        x_adv = x.copy().astype(ART_NUMPY_DTYPE)
+
+        # Compute perturbation with batching
+        for (batch_id, batch_all) in enumerate(data_loader):
+            (batch, batch_labels) = batch_all[0], batch_all[1]
+
+            batch_index_1, batch_index_2 = batch_id * self.batch_size, (batch_id + 1) * self.batch_size
+            x_adv[batch_index_1:batch_index_2] = self._generate_batch(batch, batch_labels)
+
+        return x_adv
+
+    def _generate_batch(self, x: "torch.Tensor", y: "torch.Tensor") -> np.ndarray:
+        """
+        Generate a batch of adversarial samples and return them in an array.
+
+        :param x: An array with the original inputs.
+        :param y: Target values (class labels) one-hot-encoded of shape `(nb_samples, nb_classes)`.
+        :return: Adversarial examples.
+        """
+        import torch  # lgtm [py/repeated-import]
+
+        x = x.to(self.estimator.device)
+        y = y.to(self.estimator.device)
+        x_adv = torch.clone(x)
+
+        for _ in range(self.max_iter):
+            x_adv = self._compute_torch(
+                x_adv,
+                x,
+                y,
+                self.eps_step,
+            )
```

Comment:
```suggestion
            x_adv = self._compute_torch(x_adv, x, y, self.eps_step)
```

Reply:
I didn't apply this change, because the existing version is preferred by `black==20.8b1`.